Tools/CISupport/build-webkit-org/factories.py

```diff
@@ -38,6 +38,7 @@ def __init__(self, platform, configuration, architectures, buildOnly, additional
             self.addStep(KillOldProcesses())
         self.addStep(CleanBuildIfScheduled())
         self.addStep(DeleteStaleBuildFiles())
+        self.addStep(MaybePruneCoreSymbolicationdCache())
```

Comment:
Not sure if "Maybe" adds valuable information to the step name. How is it more conditional than existing steps?

Reply:
Also, I think that this script would be executed on non-Mac queues too (Windows and Linux). We probably want to make it conditional on Mac being the host platform (keeping iOS Simulator in mind).

Reply:
A "Maybe" prefix is a naming pattern I've used in the past to describe an operation that may (or may not) cause an observable side effect. I've renamed this class to make it more clear.

I've also added a condition to this step to only run on Darwin hosts.